strapi/strapi
# [Audit logs] Process events

### What does it do?

Creates a new processEvent function inside the audit logs service, with the following responsibilities:

- Filtering events: only those in the map are handled, the others ignored
- Building the payload object based on the arguments received by the event hub
- Sanitizing the payload
- Building an object ready to be passed to the storage providers

### What does it NOT do:

- Create a provider system for the filtering. Our current idea is that the eventMap object could be extracted into a provider system in the future.
- Sanitize all the payloads. It doesn't do it for CRUD events, only for auth

### How to test it?

Do actions in Strapi, see what gets logged, see what gets ignored

co-authors @markkaylor @Feranchz
